### PR TITLE
Add trusted root ca for installations without Letsencrypt

### PR DESCRIPTION
**Pull request by corest**
Towards https://github.com/giantswarm/giantswarm/issues/14453

### CODE REVIEW

File: helm/dex-app/templates/dex-k8s-authenticator/configmap-giantswarm.yaml [hunk 1]
```diff
@@ -24,8 +27,8 @@ data:
       k8s_master_uri: https://{{ .Values.Installation.V1.Kubernetes.API.InternalAddress }}
       {{ end }}
       k8s_ca_pem: {{ toYaml .Values.Installation.V1.Secret.Kubernetes.CAPem | indent 8 }}
-      {{- if .Values.Installation.V1.GiantSwarm.OIDC.Enabled }}
```
Comment (whites11) [inline helm/dex-app/templates/dex-k8s-authenticator/configmap-giantswarm.yaml:27]: Doesn't removing the dashes cause the presence on unwanted blank lines in the generated file?

**Review by whites11 — APPROVED** [review]
LGTM when tested

File: helm/dex-app/templates/dex-k8s-authenticator/configmap-customer.yaml [hunk 2]
```diff
@@ -11,6 +11,9 @@ data:
     web_path_prefix: "/"
     debug: "false"
     logo_uri: {{ .Values.logoURI }}
+    {{ if not .Values.Installation.V1.GiantSwarm.OIDC.Letsencrypt }}
+    trusted_root_ca_file: /app/ssl/ca.crt
+    {{ end }}
```
Comment (kopiczko) [inline helm/dex-app/templates/dex-k8s-authenticator/configmap-customer.yaml:16]:
```suggestion
    {{- if not .Values.Installation.V1.GiantSwarm.OIDC.Letsencrypt }}
    trusted_root_ca_file: /app/ssl/ca.crt
    {{- end }}
```

File: helm/dex-app/templates/dex-k8s-authenticator/configmap-giantswarm.yaml [hunk 3]
```diff
@@ -10,6 +10,9 @@ data:
     web_path_prefix: "/admin"
     debug: "false"
     logo_uri: {{ .Values.logoURI }}
+    {{ if not .Values.Installation.V1.GiantSwarm.OIDC.Letsencrypt }}
+    trusted_root_ca_file: /app/ssl/ca.crt
+    {{ end }}
```
Comment (kopiczko) [inline helm/dex-app/templates/dex-k8s-authenticator/configmap-giantswarm.yaml:15]:
```suggestion
    {{- if not .Values.Installation.V1.GiantSwarm.OIDC.Letsencrypt }}
    trusted_root_ca_file: /app/ssl/ca.crt
    {{- end }}
```

**Review by kopiczko — APPROVED** [review]

**Pull request merged**